// @testing-library/react-native/build/helpers/accessiblity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.accessiblityValueKeys = exports.accessibilityStateKeys = void 0;
exports.getAccessibilityCheckedState = getAccessibilityCheckedState;
exports.getAccessibilityLabel = getAccessibilityLabel;
exports.getAccessibilityLabelledBy = getAccessibilityLabelledBy;
exports.getAccessibilityRole = getAccessibilityRole;
exports.getAccessibilityState = getAccessibilityState;
exports.getAccessibilityValue = getAccessibilityValue;
exports.getAccessibilityViewIsModal = getAccessibilityViewIsModal;
exports.isAccessibilityElement = isAccessibilityElement;
exports.isHiddenFromAccessibility = isHiddenFromAccessibility;
exports.isInaccessible = void 0;
var _reactNative = require("react-native");
var _componentTree = require("./component-tree");
var _hostComponentNames = require("./host-component-names");
const accessibilityStateKeys = ['disabled', 'selected', 'checked', 'busy', 'expanded'];
exports.accessibilityStateKeys = accessibilityStateKeys;
const accessiblityValueKeys = ['min', 'max', 'now', 'text'];
exports.accessiblityValueKeys = accessiblityValueKeys;
function isHiddenFromAccessibility(element, {
  cache
} = {}) {
  if (element == null) {
    return true;
  }
  let current = element;
  while (current) {
    let isCurrentSubtreeInaccessible = cache?.get(current);
    if (isCurrentSubtreeInaccessible === undefined) {
      isCurrentSubtreeInaccessible = isSubtreeInaccessible(current);
      cache?.set(current, isCurrentSubtreeInaccessible);
    }
    if (isCurrentSubtreeInaccessible) {
      return true;
    }
    current = current.parent;
  }
  return false;
}

/** RTL-compatitibility alias for `isHiddenFromAccessibility` */
const isInaccessible = isHiddenFromAccessibility;
exports.isInaccessible = isInaccessible;
function isSubtreeInaccessible(element) {
  // Null props can happen for React.Fragments
  if (element.props == null) {
    return false;
  }

  // See: https://reactnative.dev/docs/accessibility#aria-hidden
  if (element.props['aria-hidden']) {
    return true;
  }

  // iOS: accessibilityElementsHidden
  // See: https://reactnative.dev/docs/accessibility#accessibilityelementshidden-ios
  if (element.props.accessibilityElementsHidden) {
    return true;
  }

  // Android: importantForAccessibility
  // See: https://reactnative.dev/docs/accessibility#importantforaccessibility-android
  if (element.props.importantForAccessibility === 'no-hide-descendants') {
    return true;
  }

  // Note that `opacity: 0` is not treated as inaccessible on iOS
  const flatStyle = _reactNative.StyleSheet.flatten(element.props.style) ?? {};
  if (flatStyle.display === 'none') return true;

  // iOS: accessibilityViewIsModal or aria-modal
  // See: https://reactnative.dev/docs/accessibility#accessibilityviewismodal-ios
  const hostSiblings = (0, _componentTree.getHostSiblings)(element);
  if (hostSiblings.some(sibling => getAccessibilityViewIsModal(sibling))) {
    return true;
  }
  return false;
}
function isAccessibilityElement(element) {
  if (element == null) {
    return false;
  }
  if (element.props.accessible !== undefined) {
    return element.props.accessible;
  }
  const hostComponentNames = (0, _hostComponentNames.getHostComponentNames)();
  return element?.type === hostComponentNames?.text || element?.type === hostComponentNames?.textInput || element?.type === hostComponentNames?.switch;
}
function getAccessibilityRole(element) {
  return element.props.role ?? element.props.accessibilityRole;
}
function getAccessibilityViewIsModal(element) {
  return element.props['aria-modal'] ?? element.props.accessibilityViewIsModal;
}
function getAccessibilityLabel(element) {
  return element.props['aria-label'] ?? element.props.accessibilityLabel;
}
function getAccessibilityLabelledBy(element) {
  return element.props['aria-labelledby'] ?? element.props.accessibilityLabelledBy;
}
function getAccessibilityState(element) {
  const {
    accessibilityState,
    'aria-busy': ariaBusy,
    'aria-checked': ariaChecked,
    'aria-disabled': ariaDisabled,
    'aria-expanded': ariaExpanded,
    'aria-selected': ariaSelected
  } = element.props;
  const hasAnyAccessibilityStateProps = accessibilityState != null || ariaBusy != null || ariaChecked != null || ariaDisabled != null || ariaExpanded != null || ariaSelected != null;
  if (!hasAnyAccessibilityStateProps) {
    return undefined;
  }
  return {
    busy: ariaBusy ?? accessibilityState?.busy,
    checked: ariaChecked ?? accessibilityState?.checked,
    disabled: ariaDisabled ?? accessibilityState?.disabled,
    expanded: ariaExpanded ?? accessibilityState?.expanded,
    selected: ariaSelected ?? accessibilityState?.selected
  };
}
function getAccessibilityCheckedState(element) {
  const {
    accessibilityState,
    'aria-checked': ariaChecked
  } = element.props;
  return ariaChecked ?? accessibilityState?.checked;
}
function getAccessibilityValue(element) {
  const {
    accessibilityValue,
    'aria-valuemax': ariaValueMax,
    'aria-valuemin': ariaValueMin,
    'aria-valuenow': ariaValueNow,
    'aria-valuetext': ariaValueText
  } = element.props;
  const hasAnyAccessibilityValueProps = accessibilityValue != null || ariaValueMax != null || ariaValueMin != null || ariaValueNow != null || ariaValueText != null;
  if (!hasAnyAccessibilityValueProps) {
    return undefined;
  }
  return {
    max: ariaValueMax ?? accessibilityValue?.max,
    min: ariaValueMin ?? accessibilityValue?.min,
    now: ariaValueNow ?? accessibilityValue?.now,
    text: ariaValueText ?? accessibilityValue?.text
  };
}
